src/global-messages/GlobalMessagesContext.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import firestore, {
+  FirebaseFirestoreTypes,
+} from '@react-native-firebase/firestore';
+import {LanguageAndText} from '@atb/reference-data/types';
+import {Statuses} from '@atb/theme';
+
+export type GlobalMessage = {
+  id: string;
+  active: boolean;
+  title?: LanguageAndText[];
+  body: LanguageAndText[];
+  type: Statuses;
+  context: GlobalMessageContext[];
+};
+
+export type GlobalMessageContext =
+  | 'app-assistant'
+  | 'app-departures'
+  | 'app-ticketing'
+  | 'web-ticketing'
+  | 'web-overview';
+
+type GlobalMessageContextState = {
+  findGlobalMessages: (context: GlobalMessageContext) => GlobalMessage[];
+};
+
+const defaultGlobalMessageState = {
+  findGlobalMessages: () => [],
+};
+
+const GlobalMessagesContext = createContext<GlobalMessageContextState>(
+  defaultGlobalMessageState,
+);
+
+const GlobalMessagesContextProvider: React.FC = ({children}) => {
+  const [globalMessages, setGlobalMessages] = useState<GlobalMessage[]>([]);
+  const [error, setError] = useState(false);
+
+  useEffect(
+    () =>
+      firestore()
+        .collection('globalMessages')
+        .where('active', '==', true)
```

Comment:
Yes, we could absolutely do that. However, I guess that adding this condition means that if our non-developers inserting messages does include a single context as a string, instead of adding a single string element to an array, the message will not be fetched and displayed.

To my understanding this means that we have to choose between saving some data transfering, or be a bit more forgiving of input errors, any thoughts here? This should probably be seen in connection with the other discussion of convertors, but I think it would be beneficial to allow some errors on input instead of saving what probably is very small amounts of data. 

Remind me to remove [these lines](https://github.com/AtB-AS/mittatb-app/blob/3243b049564f3fa2fad2a1b78f6197effe47cdc1/src/global-messages/GlobalMessagesContext.tsx#L71-L73) if we opt for data saving 😅

Reply:
I would rather not allow mistypings in the schema. If it is documented as an array, we should only allow array.

Reply:
Fixed and merged: https://github.com/AtB-AS/mittatb-app/pull/2860